### ISSUE TITLE
sfdx-core/CLI: don't capitalize record types defined in scratch org def. file

### ISSUE THREAD

**Issue by manikantamullaguru** [forcedotcom/cli#2633]
> **Note** 
> Before you submit your issue, make sure that:
> - You're using the latest version of Salesforce CLI.
> - You've searched both open and closed issues for related posts.
> - You've used the `doctor` command to diagnose common issues. 
> - You understand that GitHub Issues don't adhere to any agreement or SLA.
>   - If you require immediate assistance, use official channels such as Salesforce Customer Support.

**Summary**
The Salesforce CLI capitalizes the first char of the record type defined in the scratch org def. file before deploying to the org:

**Steps To Reproduce**
1. Create a Sfdx Project.
2. Authorise a Dev hub.
3. Create a scratch org using given below scratch org definition file
{
"orgName": "MyCompany",
"edition": "Developer",
 "features": ["Communities", "ServiceCloud", "Chatbot"],
"objectSettings": {
 "case": {
    "defaultRecordType":"svc_Technical_Support"
   } } }
sf org create scratch --definition-file config/project-scratch-def.json --alias RecordTypeIssue
4.Go to the created scratch org and open the opportunity object from object manager.
5.Go to Record types and open the record type.

**Expected result**
Default record type should create with whatever user inputs in the def file.

**Actual result**
DefaultRecordType Label is creating a starting letter of word with capital.

**Specified "defaultRecordType": "svc_Technical_Support" but in the org the record type created with first letter capital - "Svc_Technical_Support"**

**Comment by cristiand391** [comment]
Follow up (and some context, this comes from a customer investigation):
CLI started doing this since we added support for defining record types in the org def. file in 2019, the API doesn't require the first char to be capitalized but changing it now would break users.

We decided to prioritize the work to introduce a config var to toggle this behavior (opt-in), then will make the switch in the future following our deprecation policy.
